errbot/backends/slack.py

```diff
@@ -568,20 +568,25 @@ def send_message(self, mess):
 
     def _slack_upload(self, stream):
         """Perform upload defined in a stream."""
-        stream.accept()
-        resp = self.api_call('files.upload', data={
-            'channels': stream.identifier.channelid,
-            'filename': stream.name,
-            'file': stream
-            })
-        if "ok" in resp and resp["ok"]:
-            stream.success()
-        else:
-            stream.error()
+        try:
+            stream.accept()
+            resp = self.api_call('files.upload', data={
+                'channels': stream.identifier.channelid,
+                'filename': stream.name,
+                'file': stream
+                })
+            if "ok" in resp and resp["ok"]:
+                stream.success()
+            else:
+                stream.error()
+        except:
+            log.error("Upload of {0} to {1} failed.".format(stream.name, stream.identifier.channelname))
```

Comment:
@jasedit Yep, that's a bit weird. Maybe it requires a fresh commit instead of amending and force pushing?